### ISSUE TITLE
Navigation - aria-expanded

### ISSUE THREAD

**Issue by lionel-m** [contao/core#7574]
Could you add the attribute `aria-expanded`to the navigation template. It is on the same level as `aria-haspopup`.

> Currently, screen reader users are unable to tell if an item has a submenu or not and if it is opened. WAI-ARIA helps to convey this information by adding the following two attributes to the menu’s HTML:

* aria-haspopup (https://github.com/contao/core/issues/6217)
* aria-expanded 

> aria-expanded is initially set to false but changed to true when the submenu opens which forces screen readers to announce that this menu item is now expanded.

[Ref.](http://www.w3.org/WAI/tutorials/menus/flyout/#improve-screen-reader-support-using-wai-aria)

**Comment by leofeyer** [comment]
@MacKP @kikmedia @NinaG Can you confirm this fix please?

**Comment by leofeyer** [comment]
@Skipman Was sagst Du dazu?

**Comment by Skipman** [comment]
Hallo Leo, 

 

ich schau mir das morgen / heute abend im detail an und geb dir dann gerne bescheid

 

LG Ronald

 

Ronald Boda

 



 

accessible.services e.U.

Aufeldsiedlung 2/1/6

3143 Pyhra

 

T +43 (664) 21 85 631

E  <mailto:info@accessible.services> info@accessible.services

I  <http://www.accessible.services/> www.accessible.services 

 

Von: Leo Feyer [mailto:notifications@github.com] 
Gesendet: Dienstag, 20. Jänner 2015 20:52
An: contao/core
Cc: Skipman
Betreff: Re: [core] Navigation - aria-expanded (#7574)

 

@Skipman <https://github.com/Skipman>  Was sagst Du dazu?

**Comment by kikmedia** [comment]
Confirmed.

**Comment by MacKP** [comment]
+1

**Comment by leofeyer** [comment]
Hm, still not sure.

> aria-expanded is initially set to false but changed to true when the submenu opens which forces screen readers to announce that this menu item is now expanded.

What does "when the submenu opens" mean? Opening a submenu is usually done via CSS or JavaScript and not via PHP. There is no way to change the attribute value via PHP when the submenu opens, because PHP will not be noticed.

I think @Skipman is right, the attribute should be used by popup menus and accordions triggering via JavaScript.

**Comment by lionel-m** [comment]
Yes via JavaScript.

**Comment by MacKP** [comment]
Hmm it could be set from Contao for an Navigation with Stoplevel = 1. But then the page reloads.. so there is no point in it as i think.
So everything is pointing that for JS. Here is an nice example: http://heydonworks.com/practical_aria_examples/#submenus 
> aia-haspopup is only added when javascript runs

**Comment by Skipman** [comment]
Da Braillezeilen und Screenreader Java-erzeugte Inhalte oft nicht optimal wiedergeben können, sollte darauf geachtet werden, Java lediglich zur Steuerung der CSS-Klassen zu verwenden. Eine - wie ich meine - gute Vorlage für die Gestaltung von Accordions mit dem ARIA - Attribut expanded liefert das Plugin **Collapse** von Bootstrap. Siehe [Bootstrap Info] (http://holdirbootstrap.de/javascript/#collapse)

**Comment by leofeyer** [comment]
I assume I can close the ticket then, since this is something which needs to be included in the JavaScript menu people are using and Contao does not include one itself?

**Comment by Skipman** [comment]
Sorry, but I don`t think so! Contao offers by itself Submenus and Accordions. For Webdesigners it isn`t nessessary, to install other Frameworks to use these Features. Contao actually offers a lot of JS Libraries and Plugins from other vendors. So it is only an improvement of accessibility, to include this features too.

**Comment by MacKP** [comment]
@Skipman i don#t think we can do it in the navigation for all. No Navigation ist like the other.. so you will have to do it for your own page. Contao didn't give you an Dropdown Navigation! A Submenu is normal. But the Submenu from the core is not dynamic. There is a Submenu or not. Nothing else. And there you don't need to mention it for Screenreaders as far as i understand this.
On the other Hand it would be nice to have it in the Accordion Element. But for that you should make a new Ticket.
regards

**Comment by Skipman** [comment]
new Tickes is open #7620

**Comment by leofeyer** [comment]
:+1:
